user-console/src/main/java/org/pentaho/mantle/client/solutionbrowser/SolutionBrowserPanel.java

```diff
@@ -438,6 +457,35 @@ public void onResponseReceived( Request request, Response response ) {
     }
   }
 
+  //TODO copy/pasted rough draft, MVP/POC implementation
+  private void openGenericFile( String filePath, COMMAND mode ) {
+    String url = null;
+    String extension = ""; //$NON-NLS-1$
+
+    //TODO Decode special characters in fileName so they dont show up URL encoded in, for example, the tab name
+    String fileName = filePath.substring( filePath.lastIndexOf( FILE_PATH_SEPARATOR ) +1 );
+    if ( fileName.lastIndexOf( FILE_EXTENSION_DELIMETER ) > 0 ) { //$NON-NLS-1$
+      extension = fileName.substring( fileName.lastIndexOf( FILE_EXTENSION_DELIMETER ) + 1 ); //$NON-NLS-1$
+    }
+    if ( !executableFileExtensions.contains( extension ) ) {
```

Comment:
Don't know if this was discussed but if the open fails for not being an executable file extension, then we should notify the user somehow. Most likely in a manner below like `showPluginError`

Reply:
I have reworked this to only accept/open the following filetypes for PVFS: ["html", "pdf", "txt", "rtf", "xlsx"](https://github.com/pentaho/pentaho-platform/pull/5616/files#diff-b95754c647cdb0e7a2f6678afd4e031dedaccedaf020d7b2b5e43f5bdcddde46R128)

If it is not a supported PVFS extension we will [show a pop-up notifying the user of this](https://github.com/pentaho/pentaho-platform/pull/5616/files#diff-b95754c647cdb0e7a2f6678afd4e031dedaccedaf020d7b2b5e43f5bdcddde46R510) much like the plugin missing error for the repository branch/similar logic